src/controller/CommissioningWindowOpener.h

```diff
@@ -107,6 +107,39 @@ class CommissioningWindowOpener
                                        Callback::Callback<OnOpenCommissioningWindow> * callback, SetupPayload & payload,
                                        bool readVIDPIDAttributes = false);
 
+    /**
+     * @brief
+     *   Try to look up the bridged device on the bridge with the given node id
+     *   of the bridge and endpoint id of the bridged device and ask it to re-enter
+     *   commissioning mode with a PASE verifier derived from the given information
+     *   and the given discriminator. The bridged device will exit commissioning mode
+     *   after a successful commissioning, or after the given `timeout` time.
+     *
+     * @param[in] deviceId      The device Id of the bridge.
+     * @param[in] endpoinId     The endpoint Id of the bridged node on the bridge.*
+     * @param[in] timeout       The commissioning mode should terminate after this much time.
+     * @param[in] iteration     The PAKE iteration count associated with the PAKE Passcode ID and ephemeral
+     *                          PAKE passcode verifier to be used for this commissioning.
+     * @param[in] discriminator The long discriminator for the DNS-SD advertisement.
+     * @param[in] setupPIN      The setup PIN to use, or NullOptional to use a randomly-generated one.
+     * @param[in] salt          The salt to use, or NullOptional to use a
+     *                          randomly-generated one.  If provided, must be at
+     *                          least kSpake2p_Min_PBKDF_Salt_Length bytes and
+     *                          at most kSpake2p_Max_PBKDF_Salt_Length bytes in
+     *                          length.
+     * @param[in] callback      The function to be called on success or failure of opening of commissioning window.
+     * @param[out] payload      The setup payload, not including the VID/PID bits,
+     *                          even if those were asked for, that is generated
+     *                          based on the passed-in information.  The payload
+     *                          provided to the callback function, unlike this
+     *                          out parameter, will include the VID/PID bits if
+     *                          readVIDPIDAttributes is true.
+     *
+     */
+    CHIP_ERROR OpenCommissioningWindow(NodeId deviceId, EndpointId endpointId, System::Clock::Seconds16 timeout, uint32_t iteration,
```

Comment:
These methods start taking too many arguments. Could you create some 'Params' class instead? Methods with more than 3-4 arguments are really hard to follow. This has 10. Same for below.